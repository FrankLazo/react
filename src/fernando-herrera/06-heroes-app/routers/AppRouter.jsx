import { useContext } from 'react';
import { BrowserRouter as Router, Switch } from 'react-router-dom';
import AuthContext from '../auth/AuthContext';
import LoginScreen from '../components/login/LoginScreen';
import DashboardRoutes from './DashboardRoutes';
import PrivateRoute from './PrivateRoute';
import PublicRoute from './PublicRoute';

const AppRouter = () =>
{
	const { user: { logged } } = useContext(AuthContext);

	return (
		<Router>
			<div className="min-height-full has-background-light">
				<Switch>
					<PublicRoute exact path="/login" component={ LoginScreen }  isAuthenticated={ logged } />
					<PrivateRoute path="/" component={ DashboardRoutes } isAuthenticated={ logged } />
				</Switch>
			</div>
		</Router>
	);
};

export default AppRouter;
