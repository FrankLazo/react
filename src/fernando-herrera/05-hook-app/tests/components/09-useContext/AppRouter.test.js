import { mount } from 'enzyme';
import AppRouter from '../../../components/09-useContext/AppRouter';
import UserContext from '../../../components/09-useContext/UserContext';

describe('<AppRouter /> tests', () => {
	const user = {
		name: 'Frank',
		mail: 'example@mail.com',
	};

	const wrapper = mount(
		<UserContext.Provider value={{ user }}>
			<AppRouter />
		</UserContext.Provider>
	);

	test('should show the right component', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
