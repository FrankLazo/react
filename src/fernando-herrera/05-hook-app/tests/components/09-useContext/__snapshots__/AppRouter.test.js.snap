// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppRouter /> tests should show the right component 1`] = `
<div>
  <nav
    className="navbar"
  >
    <div
      className="navbar-brand"
    >
      <a
        className="navbar-item has-text-weight-bold"
        href="/"
        onClick={[Function]}
      >
        useContext
      </a>
    </div>
    <div
      className="navbar-menu is-active"
    >
      <ul
        className="navbar-start"
      >
        <li
          className="navbar-item"
        >
          <a
            aria-current="page"
            className="has-text-primary"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
        </li>
        <li
          className="navbar-item"
        >
          <a
            aria-current={null}
            href="/About"
            onClick={[Function]}
          >
            About
          </a>
        </li>
        <li
          className="navbar-item"
        >
          <a
            aria-current={null}
            href="/Login"
            onClick={[Function]}
          >
            Login
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <div
    className="min-height-full content has-background-dark p-6 has-text-light"
  >
    <h1
      className="has-text-light"
    >
      Home
    </h1>
    <hr
      className="full-rule has-background-grey-dark"
    />
    <pre>
      {
  "name": "Frank",
  "mail": "example@mail.com"
}
    </pre>
  </div>
</div>
`;
